core/include/cubos/input/input_sources/single_axis.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef CUBOS_INPUT_SINGLE_AXIS_HPP
+#define CUBOS_INPUT_SINGLE_AXIS_HPP
+
+#include <cubos/io/window.hpp>
+#include <cubos/input/input_sources/input_source.hpp>
+#include <variant>
+#include <tuple>
+
+namespace cubos::input
+{
+
+    class SingleAxis : public InputSource
+    {
+
+    public:
+        SingleAxis(cubos::io::MouseAxis axis);
+        SingleAxis(cubos::io::Key negativeKey, cubos::io::Key positiveKey);
```

Comment:
I was trying to think the structure of the single axis constructor and this seems like the obvious way but should't we allow for other types of inputs? Keyboards, joysticks, controllers, etc...
And if so would it make sense to allow combining them? e.g posAxis is binded to a keyboard Key and the negAxis to a mouse button.
This would probably make a nightmare for constructors, I don't know if we can use templates or variants for this.
Thoughts?

Reply:
The constructor will always need to be there. Either in the same class SingleAxis or in multiple classes SingleJoystickActions, SingleMouseAxis. 

The only problem might be associated with the number of handle functions that need to be created for supporting multiple controls. However, has the output of the controls stays the same (e.g. button being pressed, axis changing a value) we will be fine. Even if we split the multiple axis supported across multiple implementations, we would always need to do multiple handles.

I prefer this way because it is clear to the game developer what each input source produces regardless of the controls that produced it.